### PR TITLE
Add ServerMessage.termination handler

### PR DESCRIPTION
**Pull request by nbsp**
untested, SFU hasn't implemented JobTermination yet

### CODE REVIEW

File: livekit-agents/livekit/agents/worker.py [hunk 1]
```diff
@@ -615,3 +617,17 @@ def _handle_assignment(self, assignment: agent.JobAssignment):
             logger.warning(
                 "received assignment for an unknown job", extra={"job": assignment.job}
             )
+
+    async def _handle_termination(self, msg: agent.JobTermination):
+        proc = next(
+            (
+                x
+                for x in self._proc_pool.processes
```
Comment (theomonnom) [inline livekit-agents/livekit/agents/worker.py:625]: Maybe we should introduce a map<job_id, proc> inside the proc_pool?

File: livekit-agents/livekit/agents/ipc/proc_pool.py [hunk 2]
```diff
@@ -48,6 +48,16 @@ def __init__(
     def processes(self) -> list[SupervisedProc]:
         return self._processes
 
+    def get(self, job_id: str) -> SupervisedProc | None:
+        return next(
```
Comment (theomonnom) [inline livekit-agents/livekit/agents/ipc/proc_pool.py:52]: Sorry, by a map I mean to avoid iterating, and creating a dict instead

Comment (nbsp) [reply livekit-agents/livekit/agents/ipc/proc_pool.py:52]: i don't think it makes sense for us here, the proc is created and referred to before a job is assigned to it

File: livekit-agents/livekit/agents/worker.py [hunk 3]
```diff
@@ -485,6 +485,8 @@ async def _recv_task():
                     self._handle_availability(msg.availability)
                 elif which == "assignment":
                     self._handle_assignment(msg.assignment)
+                elif which == "termination":
+                    await self._handle_termination(msg.termination)
```
Comment (theomonnom) [inline livekit-agents/livekit/agents/worker.py:489]: This needs to be in a background task (the process can take 60s to shutdown)

Put it inside self._tasks

File: livekit-agents/livekit/agents/ipc/proc_pool.py [hunk 4]
```diff
@@ -48,6 +48,16 @@ def __init__(
     def processes(self) -> list[SupervisedProc]:
         return self._processes
 
+    def get(self, job_id: str) -> SupervisedProc | None:
```
Comment (theomonnom) [inline livekit-agents/livekit/agents/ipc/proc_pool.py:51]:
```suggestion
    def get_by_job_id(self, job_id: str) -> SupervisedProc | None:
```

File: livekit-agents/livekit/agents/worker.py [hunk 5]
```diff
@@ -485,6 +485,12 @@ async def _recv_task():
                     self._handle_availability(msg.availability)
                 elif which == "assignment":
                     self._handle_assignment(msg.assignment)
+                elif which == "termination":
+                    user_task = self._loop.create_task(
+                        self._handle_termination(msg.termination), name="termination"
```
Comment (theomonnom) [inline livekit-agents/livekit/agents/worker.py:490]: nit
```suggestion
                        self._handle_termination(msg.termination), name="agent_job_termination"
```

**Review by theomonnom — APPROVED** [review]

**Pull request merged**